subprojects/functional/src/main/java/org/gradle/internal/lazy/UnsafeLazy.java

```diff
@@ -15,11 +15,13 @@
  */
 package org.gradle.internal.lazy;
 
+import javax.annotation.Nullable;
 import javax.annotation.concurrent.NotThreadSafe;
 import java.util.function.Supplier;
 
 @NotThreadSafe
 class UnsafeLazy<T> implements Lazy<T> {
+    @Nullable
```

Comment:
🧐 Should this be added to the constructor arg as well?

Reply:
Perhaps it would be better not to include the annotation 🤔

Reply:
Or maybe leave it and explicitly say `@Nonnull` on the constructor if that's the rule?

While we're at it, `value` is definitely nullable as well, at least at first.  Is it implied that the `Supplier` here should return a non-null resul?

Reply:
Supplier should always return nkn null, yes. We should probably document that.

Reply:
It should be ok for the Supplier to return null as long as null is not treated as a special value.